### ISSUE TITLE
Alligator example

### ISSUE THREAD

**Issue by Daniel-Gorman** [brianstock/MixSIAR#149]
Hi Brian, fairly new user of MixSIAR. I am trying to run the script for the recently included Alligator example through R Studio, but I don’t seem to be able to get past line 173. This then appears to mean that I cannot subsequently ‘compare_models’ (screenshot below). Apologies if it’s a naively simple issue, but any advice would be greatly appreciated. 
 
![image](https://user-images.githubusercontent.com/42760611/44699932-1e5ec880-aab9-11e8-9cfa-c95183590a02.png)

Attached also is Sessioninfo():
 
![image](https://user-images.githubusercontent.com/42760611/44699949-2ae32100-aab9-11e8-9c48-4b6f4c2e2102.png)

**Comment by brianstock** [comment]
Hi Daniel,

I just re-ran `mixsiar_script_alligator.R` on my machine to confirm it is working.

My guess is you're getting an error in the for loop with the call to `plot_data` (since it's graphics-related) and then not actually running the models. If you look at each element of the `jags.mod` list, you should see `rjags` objects. I.e. if you run `class(jags.mod[[1]])`, the output should be `[1] "rjags"`.

Assuming this is the case, set `mod=1` and then go through the for loop line-by-line to see what is causing the error. My guess is that it is related to graphics. Sometimes R Studio causes issues here, so first thing I'd try would be using the R console instead.

**Comment by Daniel-Gorman** [comment]
Hi Brian, Ive checked the objects - and get [1] = NULL (rather than "rjags").. ?
Also, I'm not quite sure what you mean by "go through the for loop line-by-line".. 
Thanks for any additional help. 

![image](https://user-images.githubusercontent.com/42760611/44764662-6b0ed600-ab83-11e8-9b63-eb991b49e25e.png)

**Comment by brianstock** [comment]
Yeah, that's what I expected - you're getting an error before running the model (there is no model object saved as `jags.mod[[1]]`). After setting `mod=1`, run the lines inside the for loop (i.e. 109-166) one command at a time so you can see where the error occurs.

**Comment by Daniel-Gorman** [comment]
Hi Brian, slowly getting through the examples. Managed to fix the previous issue by changing the for loop section:

  plot_data(filename=paste0("isospace_plot_", mod),
            plot_save_pdf=FALSE,
   
I'm now using my own data (with 4 competing models), but getting the following error when moving back up to the root directory:
![image](https://user-images.githubusercontent.com/42760611/44782815-125b2f80-abbb-11e8-896f-34bec1d8fd0d.png)

**Comment by brianstock** [comment]
Good to hear! Saving PDFs is likely an OS-specific issue. Did you try using the R console instead of R Studio?

The new error is happening reading in the source CSV file, not when moving back to the root directory. It's probably worth a bit of your time to do an R tutorial and understand what a for loop is. Cyclismo is pretty good: https://www.cyclismo.org/tutorial/R/scripting.html#for-statements.

**Comment by Daniel-Gorman** [comment]
Hi again Brian, making steady progress now :) Thanks for the very responsive help.. 
Ive made it all the way through running my 4 models (Null, factor, covariate, factor+covariate) and used LOO to compare the weights!
Now trying to plot the covariate (similar to the Alligator plot) but bumped into some more issues. Any advice on the following error?
![image](https://user-images.githubusercontent.com/42760611/44858749-4579ed80-aca5-11e8-92a2-f1e8ffa48d0c.png)

**Comment by brianstock** [comment]
Sorry, this error is sloppy coding/testing on my part. The `calc_eps` function should be moved above where it is called. I've fixed this on the develop branch, which you can install via
```
devtools::install_github("brianstock/MixSIAR", ref="develop")
```
Alternatively, you can just copy+paste the `calc_eps` function (lines 274-280), then re-run that code.

**Comment by Daniel-Gorman** [comment]
Hi Brian - this worked - thanks greatly. 
So, now I’m almost where I want to be, but I fear that the "hard part" is yet to come. My covariate (‘distance’) model includes 6 dietary sources (which makes interpretation very difficult). What I would like to do is combine the 6 sources – into 3 groups (a posteriori).. 
Something like this (taken from the Mantis e.g.):

combined <- combine_sources(jags.model, mix, source, alpha.spec, 
                            groups=list(group1=c("source1","source2"), group2=c("source3","source4"), group3=c("source5","source6")))

..and then Plot [group] proportions vs. ‘distance’ from covariate model. 
I'm currently trying to merge the code from Mantis (combine) + Alligator (plot proportions vs. length) > but not getting very far :0
Any advice on how to move forward, greatly appreciated.

**Comment by bggalvenmon** [comment]
Hi Brian, 
I also am having trouble with the alligator example. I had the same problem as Daniel first, and I was able to fix the problem with Daniels solution. However, after running the JAGS model and getting the density plots, I am getting an error when I try to run line 171 (comparison.table). 

I think that it could be two things: 1) I might be having a problem with the for-loop as I am unable to run the loop from line 107-167 in one command. When I hit ctrl+enter on line 107, the console spits out everything but the only code that seems to actually run is the isospace plot...the JAGS model does not run. However, if I run the code one command at a time, the model runs and works fine and the posterior density plots are created as well. 
2) Another problem could be that my jags.mods are all NULL except the first one. Looking at my jags.mod vector, after setting mod=1, the rest of my models (2-8) are all NULL. 

I don't know why this is happening, and I have restarted R, downloaded the develop branch and reinstalled mixSIAR. I also tried running the for loop in the console with no success in getting the JAGS model to run. I would greatly appreciate the help as I am trying to using this type of analysis for my masters thesis. Thanks a lot Brian, hope to hear from you soon. 

Here is the error: 

> comparison.table <- compare_models(jags.mod)
Error in nobs[m] <- dim(x[[m]]$BUGSoutput$sims.list$loglik)[2] : 
  replacement has length zero
In addition: Warning message:
Relative effective sample sizes ('r_eff' argument) not specified.
For models fit with MCMC, the reported PSIS effective sample sizes and 
MCSE estimates will be over-optimistic.
